### PR TITLE
Rfc80/clinical data counts refactor

### PR DESCRIPTION
**Pull request by haynescd**
Refactor Clinical data counts to use SQL only

### CODE REVIEW

**Review by fuzhaoyuan — COMMENTED** [review]
Do you think if it's possible to change from using [ClinicalDataBinner](https://github.com/cBioPortal/cbioportal/blob/ca357e46f294e164455029c79fc4946d652285ea/src/main/java/org/cbioportal/web/columnar/ClinicalDataBinner.java#L45) to [BasicDataBinner](https://github.com/cBioPortal/cbioportal/blob/ca357e46f294e164455029c79fc4946d652285ea/src/main/java/org/cbioportal/web/columnar/BasicDataBinner.java#L61)?

From @dippindots it's usable. And then we can get rid of `ClinicalDataBinner`.

**Review by fuzhaoyuan — DISMISSED** [review]
lgtm

File: src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewMapper.xml [hunk 1]
```diff
@@ -165,63 +169,63 @@
...
-
-    <sql id="getClinicalDataCountsQuerySample">
+    <sql id="getClinicalDataCountsQuery">
+        (
+        WITH clinical_data_query AS (
         SELECT
-        attribute_name as attributeId,
-        <include refid="normalizeAttributeValue">
-            <property name="attribute_value" value="attribute_value"/>
-        </include> as value,
```
Comment (onursumer) [inline src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewMapper.xml:174]: Don't we want to normalize NA values (like N/A, NAN, etc.) anymore? Did we completely remove them from the database?

File: src/test/java/org/cbioportal/persistence/mybatisclickhouse/StudyViewMapperClinicalDataCountTest.java [hunk 2]
```diff
@@ -139,20 +160,26 @@ public void getAgeCountsForMultipleStudies() {
...
+
+        assertTrue(ageCountsOptional.isPresent());
+        var ageCounts = ageCountsOptional.get().getCounts();
+        
         // everything should be exactly the same as single study (STUDY_GENIE_PUB) filter
         // except NA counts
         assertAgeCounts(ageCounts);
 
         // TODO this fails because of a known issue 
         //  (https://github.com/cBioPortal/rfc80-team/issues/39)
```
Comment (onursumer) [inline src/test/java/org/cbioportal/persistence/mybatisclickhouse/StudyViewMapperClinicalDataCountTest.java:180]: we should also remove this TODO

**Review by onursumer — DISMISSED** [review]
Looks good. Just commented on a few minor things.

File: src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewMapper.xml (hunk 1 above)
Comment (haynescd) [reply src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewMapper.xml:174]: Yea, weird how it shows it removed. I have it here in line 181

`            AND 
            <include refid="normalizeAttributeValue">
                <property name="attribute_value" value="value"/>
            </include>`

**Pull request merged**